worker.go

```diff
@@ -49,12 +50,18 @@ type (
 		// default: defaultMaxConcurrentActivityExecutionSize(1k)
 		MaxConcurrentActivityExecutionSize int
 
-		// Optional: Sets the rate limiting on number of activities that can be executed per second.
+		// Optional: Sets the rate limiting on number of activities that can be executed per refresh duration.
 		// This can be used to protect down stream services from flooding.
 		// The zero value of this uses the default value.
 		// default: defaultMaxActivityExecutionRate(100k)
 		// Warning: activity's StartToCloseTimeout starts ticking even if a task is blocked due to rate limiting.
-		MaxActivityExecutionsPerSecond float32
+		MaxActivityExecutionRate int
+
+		// Optional: Sets the refresh duration for rate limit. If not specified, it uses 1s as default.
+		// Use this to fine tune your rate limiter. For example, you could set this duration to 10s with max rate set to
+		// 1 which means we rate limit the activity task to at most 1 per every 10s. You could also set this duration to
+		// 100ms with max rate set wo 10 which means rate limit to 10 activity task execution per every 100ms.
+		MaxActivityExecutionRateRefreshDuration time.Duration
```

Comment:
using 2 variables of rate and refill duration gives user more precise control of the behavior, and clear behavior expectation. It is unclear what it means for 1.1 RPS if we use only one float variable for rate.